code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -916,9 +930,60 @@ Pass a positive integer as an argument to override a bot's default speed.
 /mob/living/simple_animal/bot/sentience_act()
 	faction -= "silicon"
 
-/mob/living/simple_animal/bot/proc/activate_data_hud()
-//If a bot has its own HUD (for player bots), provide it.
-	if(!data_hud_type)
+/mob/living/simple_animal/bot/proc/set_path(list/newpath)
+	path = newpath ? newpath : list()
+	if(!path_hud)
+		return
+	var/list/path_huds_watching_me = list(GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED])
+	if(path_hud)
+		path_huds_watching_me += path_hud
+	for(var/V in path_huds_watching_me)
+		var/datum/atom_hud/H = V
+		H.remove_from_hud(src)
+
+	var/list/path_images = hud_list[DIAG_PATH_HUD]
+	if(path_images.len)
+		for(var/V in path_images)

```

Comment:
you should try to reuse them.

Reply:
I am not going to attempt making an image pool in this PR.